services/webhooks2tasks/src/handlers/problems/harborScanningCompleted.ts

```diff
@@ -76,29 +76,38 @@ const DEFAULT_REPO_DETAILS_MATCHER = {
     const environmentResult = await getEnvironmentByOpenshiftProjectName(openshiftProjectName);
     const environmentDetails: any = R.prop('environmentByOpenshiftProjectName', environmentResult)
 
+    let size = 20; let batchedArrVulns = [];
+    if (vulnerabilities) {
+      for (let i = 0; i < vulnerabilities.length; i += size) {
+        batchedArrVulns.push(vulnerabilities.slice(i, i + size));
+      }
+    }
 
-    let messageBody = {
-      lagoonProjectId,
-      lagoonProjectName,
-      lagoonEnvironmentId: environmentDetails.id,
-      lagoonEnvironmentName: environmentDetails.name,
-      lagoonServiceName,
-      vulnerabilities,
-    };
-
-    const webhookData = generateWebhookData(
-      WebhookRequestData.giturl,
-      'problems',
-      'harbor:scanningresultfetched',
-      messageBody
-    );
-
-    const buffer = new Buffer(JSON.stringify(webhookData));
-
-    await channelWrapperWebhooks.publish(`lagoon-webhooks`, '', buffer, {
-      persistent: true,
-    });
-
+    if (batchedArrVulns) {
```

Comment:
Hey @timclifford - I don't know about this logic here, I think it introduces an error.
The problem is that when we get a new set of vulnerabilities - we actually clear out the current set of vulnerabilities for that environment (https://github.com/amazeeio/lagoon/blob/9e64343280ac99e0af5f5c4d7b144a2aff149781/services/webhooks2tasks/src/handlers/problems/processHarborVulnerabilityList.ts#L37)

If I'm reading this correctly, we're splitting a set of vulns into multiple sets - but doesn't this mean that everytime we process a new chunked set of vulnerabilities, the last set is deleted _even if that set is part of the current vunerability scan_? (i.e. if there are three chunks, only the problems from the 3rd chunk will show up?)

Reply:
@bomoko ah yes - I thought I had this chunking post deletion of problems from source, but you are right this would be an issue. I was thinking though that rather than change this slightly, the best solution would be to be replace this with the change to the api we discussed. Which would be to create a bulkProblems mutation in the api that we would use here (instead of iterating through each problem and adding that single problem via the addProblem query). Then we could also control this issue within the resolver of the api itself (which seems like the better place for it). Ideally, the issue we have lies in the fact that keycloak is timing out/has too many open connections due to being overloaded by the api so makes sense to actually try tackle that there.

Reply:
so i suggest perhaps we hold this - and revise once we have considered how the api should handle large volumes of problems like this.